tests/wasm/runtime/abi.ts

```diff
@@ -35,7 +35,7 @@ export class Abi {
   }
 
   private readArray<T>(ptr: Pointer, reader: (value: Pointer) => T): T[] | null {
-    if (ptr === null) {
+    if (ptr === 0) {
```

Comment:
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia0.giphy.com%2Fmedia%2FSXrX5jIGoeZxhDMHLf%2Fgiphy.gif&token=sigh" data-gitmeme-token="sigh"><img src="https://media0.giphy.com/media/SXrX5jIGoeZxhDMHLf/giphy.gif" title="Created by gitme.me with /sigh"/></a>

This should be caught by the typechecker (cf [open issue](https://github.com/microsoft/TypeScript/issues/11920)), maybe we can use a lint rule for now to catch these: https://github.com/microsoft/TypeScript/issues/11920https://palantir.github.io/tslint/rules/strict-type-predicates/

Reply:
We're using ESLint-TypeScript (since, from what I understand, `tslint` is deprecated) and I couldn't find the rule at first sight: https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/README.md#extension-rules

Reply:
I think `@typescript-eslint/no-unnecessary-condition` would have caught this (https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unnecessary-condition.md).

Reply:
:detective: